app/src/main/java/com/epfl/drawyourpath/database/FirebaseDatabase.kt

```diff
@@ -382,6 +463,200 @@ class FirebaseDatabase : Database() {
+        return future
+    }
+
+    override fun addChatMember(userId: String, conversationId: String): CompletableFuture<Unit> {
+        val future = CompletableFuture<Unit>()
+        // add the new member to the members list
+        val newMember = HashMap<String, Any>()
+        newMember.put(userId, true)
+        chatMembers(conversationId).updateChildren(newMember).addOnSuccessListener {
+            // add the chat to the chat list of the user with userId
+            val newChat = HashMap<String, Any>()
+            newChat.put(conversationId, true)
+            userProfile(userId).child(FirebaseKeys.USER_CHATS).updateChildren(newChat)
+                .addOnSuccessListener { future.complete(Unit) }
+                .addOnFailureListener { future.completeExceptionally(it) }
+        }.addOnFailureListener { future.completeExceptionally(it) }
+        return future
+    }
+
+    override fun removeChatMember(userId: String, conversationId: String): CompletableFuture<Unit> {
+        val future = CompletableFuture<Unit>()
+        // remove the member to the member chat list
+        chatMembers(conversationId).child(userId).removeValue()
+            .addOnSuccessListener {
+                // remove the chat to the user with userId chat list
+                userProfile(userId).child(FirebaseKeys.USER_CHATS).child(conversationId)
+                    .removeValue()
+                    .addOnSuccessListener { future.complete(Unit) }
+                    .addOnFailureListener { future.completeExceptionally(it) }
+            }.addOnFailureListener { future.completeExceptionally(it) }
+        return future
+    }
+
+    override fun getChatMessages(conversationId: String): CompletableFuture<List<Message>> {
+        val future = CompletableFuture<List<Message>>()
+        chatMessages(conversationId).get().addOnSuccessListener { data ->
+            val listMessage = ArrayList<Message>()
+            // Log.println(Log.INFO,"", data.children.toMutableList().get(0).toString())
+            for (elem in data.children) {
+                listMessage.add(getMessageFromData(elem))
+            }
+            future.complete(listMessage)
+        }.addOnFailureListener { future.completeExceptionally(it) }
+        return future
+    }
+
+    override fun addChatMessage(conversationId: String, message: Message): CompletableFuture<Unit> {
+        // add the message to the list of the messages of the conversation
+        if (message.content.javaClass == MessageContent.RunPath::class.java) {
+            return addChatRunMessage(conversationId, message)
+        }
+        if (message.content.javaClass == MessageContent.Picture::class.java) {
+            return addChatPictureMessage(conversationId, message)
+        }
+        if (message.content.javaClass == MessageContent.Text::class.java) {
+            return addChatTextMessage(conversationId, message)
+        }
+        val future = CompletableFuture<Unit>()
```

Comment:
Suggestion: The current implementation of addChatMessage uses a series of if statements to determine the type of message content. This approach may become cumbersome as you add more content types. Instead, consider using a more scalable approach like the visitor pattern or using Kotlin's "when" statement for better code readability and maintainability.